src/achievements.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#include "achievements.hpp"
+
+#include "filesystem.hpp"
+#include "game_config.hpp"
+#include "log.hpp"
+#include "preferences/general.hpp"
+#include "serialization/parser.hpp"
+#include "serialization/preprocessor.hpp"
+
+static lg::log_domain log_config("config");
+#define ERR_CONFIG LOG_STREAM(err, log_config)
+
+// TODO: horizontal scrollbar is present despite `horizontal_scrollbar_mode = "never"`
+// TODO: how to make the initial height more than a single achievement? hard code the height? add more achievements?
```

Comment:
I think there's a macro to make a widget take up a minimum size? It essentially stacks it on top of a spacer.

Reply:
This has been completed - the macro is `GUI_FORCE_WIDGET_MINIMUM_SIZE`.